### DISCUSSION TITLE
How to assign relations from incoming API?

### DISCUSSION THREAD

**Discussion by Morzaram** [flutterdata/flutter_data#204] [Q&A]
Hey there I'm trying to associate my data from my incoming save() but for some reason, it's not associating. I'm trying to understand what is happening from the docs, but I'm not following. 

I saw this, but there weren't any details: https://github.com/flutterdata/flutter_data/issues/140#issuecomment-1046384080

It was saying that it will associate it if I give it the voice_id, but it wasn't.

Only when I refetch the data by going back a page will it fetch the associated voice.

My incoming data for Thought.save() is: 
![image](https://user-images.githubusercontent.com/70202379/219012130-e998881d-d501-49cf-8777-7c58eb1dedaf.png)

Thought model
```dart
@JsonSerializable()
@DataRepository([UrlAdapter, ThoughtAdapter])
class Thought extends DataModel<Thought> {
  @override
  final String? id;
  final String message;
  final String? createdAt;
  final BelongsTo<Voice> voice;
  final BelongsTo<Topic> topic;
  Thought({
    this.id,
    this.createdAt,
    required this.voice,
    required this.topic,
    required this.message,
  });
}
```

Save thought function
```dart
 await Thought(
              message: message,
              topic: BelongsTo(widget.topic),
              voice: BelongsTo(currentVoice))
          .save();
```

Am I misunderstanding that it's possible to do this? Am I not calling something?

That being said, having a 'voice' key with the associated voice_id will work.

**Comment by frank06** [comment]
It's definitely possible. You can step through this function: https://github.com/flutterdata/flutter_data/blob/master/lib/src/repository/remote_adapter.dart#L629

So you're saying this only happens when you get a response for `save()` but not from a finder, is my understanding correct?

How are you checking this? Are you sure this doesn't have to do with whether the widget is repainted?

**Reply by Morzaram** [reply]
I will definitely recheck this now that I have more experience. This could definitely be the case. Thank you for always responding ☺️
